pkg/crcontrollers/node/common/annotation.go

```diff
@@ -22,81 +22,127 @@ import (
 	"strings"
 	"time"
 
+	"github.com/dell/csi-baremetal/api/v1/nodecrd"
 	"github.com/sirupsen/logrus"
 	corev1 "k8s.io/api/core/v1"
 	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/dell/csi-baremetal/pkg/base/featureconfig"
 )
 
-const (
-	// NumberOfRetries to obtain Node ID
-	NumberOfRetries = 20
-	// DelayBetweenRetries to obtain Node ID
-	DelayBetweenRetries = 3
-)
+type service struct {
+	client            k8sClient.Client
+	log               *logrus.Logger
+	featureConfig     featureconfig.FeatureChecker
+	delayBetweenRetry time.Duration
+	numberOfRetry     int
+}
 
-// ObtainNodeIDWithRetries obtains Node ID with retries
-func ObtainNodeIDWithRetries(client k8sClient.Client, featureConf featureconfig.FeatureChecker, nodeName string,
-	nodeIDAnnotation string, logger *logrus.Logger, retries int, delay time.Duration) (nodeID string, err error) {
-	// try to obtain node ID
-	for i := 0; i < retries; i++ {
-		logger.Info("Obtaining node ID...")
-		if nodeID, err = GetNodeIDByName(client, nodeName, nodeIDAnnotation, "", featureConf); err == nil {
-			logger.Infof("Node ID is %s", nodeID)
-			return nodeID, nil
-		}
-		logger.Warningf("Unable to get node ID due to %v, sleep and retry...", err)
-		time.Sleep(delay * time.Second)
+// NodeAnnotation represent annotation service
+type NodeAnnotation interface {
```

Comment:
Looks like it doesn't depend on annotation anymore. Should we call this interface `NodeUUID` or something like this?